### PR TITLE
chore: SEO and a11y improvements

### PR DESCRIPTION
**Pull request by siddharthkul**
**What changed? Why?**
- Added page level titles and descriptions for SEO 🌐
- Improved A11y Lighthouse score by adding ariaLabels where needed 🚀
- Extended support for ariaLabel prop for NextLink component 🧩

Screenshot 2024-02-23 at 10 44 48 AM

**How has it been tested?**
✅ Tested locally
✅ Lighthouse chrome plugin

### CODE REVIEW

File: template/web/app/page.tsx [hunk 1]
```diff
@@ -2,8 +2,9 @@ import { generateMetadata } from '@/utils/generateMetadata';
 import Home from './home';
 
 export const metadata = generateMetadata({
-  title: 'Build Onchain Apps Toolkit',
-  description: 'Build Onchain Applications with the best consumer experience in a few minutes.',
+  title: 'App - BOAT',
```
Comment (Zizzamia) [inline template/web/app/page.tsx:5]: Maybe this one can be just `Build Onchain Apps Toolki` as this is the homepage.

File: template/web/app/paymaster-bundler/layout.tsx [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import { generateMetadata } from '@/utils/generateMetadata';
 
 export const metadata = generateMetadata({
-  title: 'Build Onchain Apps Toolkit',
-  description: 'Build Onchain Applications with the best consumer experience in a few minutes.',
+  title: 'Paymaster - BOAT',
```
Comment (Zizzamia) [inline template/web/app/paymaster-bundler/layout.tsx:4]: Let's do `Paymaster Bundler`

**Review by Bassim23 — APPROVED** [review]

File: template/web/app/paymaster-bundler/layout.tsx (hunk 2 above)
Comment (kyhyco) [reply template/web/app/paymaster-bundler/layout.tsx:4]: @Zizzamia should we be spelling out `BOAT` or keep it as an acronym?

File: yarn.lock [hunk 3]
```diff
@@ -22,7 +22,7 @@ __metadata:
   languageName: node
   linkType: hard
 
-"@babel/code-frame@npm:^7.0.0, @babel/code-frame@npm:^7.22.13, @babel/code-frame@npm:^7.23.5":
+"@babel/code-frame@npm:^7.0.0, @babel/code-frame@npm:^7.23.5":
```
Comment (Zizzamia) [inline yarn.lock:25]: Why we got a `yarn` change?

Comment (Zizzamia) [reply yarn.lock:25]: I think is best we keep it out of this one.

File: template/web/src/components/code-block/CodeBlock.module.css [hunk 4]
```diff
@@ -12,7 +12,7 @@
   z-index: 2;
   top: 2px;
   right: 8px;
-  color: rgb(235 235 245 / 0.38);
+  color: rgb(235 235 245 / 0.75);
```
[the remark this answers is not in the document]
Comment (siddharthkul) [reply template/web/src/components/code-block/CodeBlock.module.css:15]: Before
Screenshot 2024-02-23 at 11 07 25 AM

After
Screenshot 2024-02-23 at 11 07 40 AM

This helps contrast which bumps up Lighthouse score.

[the remark this answers is not in the document]
Comment (siddharthkul) [reply template/web/src/components/code-block/CodeBlock.module.css:15]: The color change is applied on "sh", top-right of codeblock.

File: yarn.lock (hunk 3 above)
Comment (siddharthkul) [reply yarn.lock]: removed

File: template/web/src/components/code-block/CodeBlock.module.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (Zizzamia) [reply template/web/src/components/code-block/CodeBlock.module.css:15]: I will rather keep it as it is, as the goal is to have it low contrast. And have low Lighthouse is cool to me. The main goal is the codeblock to feel slick.

[the remark this answers is not in the document]
Comment (siddharthkul) [reply template/web/src/components/code-block/CodeBlock.module.css:15]: Sounds good.

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**